### PR TITLE
feat: rename Impure to Univ

### PR DESCRIPTION
**Pull request by mlutze**
related to #7114

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/api/lsp/provider/HoverProvider.scala [hunk 1]
```diff
@@ -176,7 +176,7 @@ object HoverProvider {
 
     val p = eff0 match {
       case Type.Cst(TypeConstructor.Pure, _) => ""
-      case Type.Cst(TypeConstructor.EffUniv, _) => raw" \ IO"
+      case Type.Cst(TypeConstructor.Univ, _) => raw" \ IO"
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/HoverProvider.scala:179]: Hover as IO?

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/CompletionUtils.scala [hunk 2]
```diff
@@ -73,7 +73,7 @@ object CompletionUtils {
 
       val eff = eff0 match {
         case Type.Cst(TypeConstructor.Pure, _) => ""
-        case Type.Cst(TypeConstructor.EffUniv, _) => raw" \ IO"
+        case Type.Cst(TypeConstructor.Univ, _) => raw" \ IO"
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/CompletionUtils.scala:76]: IO?

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/InstanceCompleter.scala [hunk 3]
```diff
@@ -74,7 +74,7 @@ object InstanceCompleter extends Completer {
       val retTpe = fmtType(clazz, sig.spec.retTpe, hole)
       val eff = sig.spec.eff match {
         case Type.Cst(TypeConstructor.Pure, _) => ""
-        case Type.Cst(TypeConstructor.EffUniv, _) => raw" \ IO"
+        case Type.Cst(TypeConstructor.Univ, _) => raw" \ IO"
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/InstanceCompleter.scala:77]: IO?

File: main/src/ca/uwaterloo/flix/api/lsp/provider/HoverProvider.scala (hunk 1 above)
Comment (mlutze) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/HoverProvider.scala:179]: Good catch

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**